File: src/transformers/features/http/CreateHttpHeadersTransformer.ts
import { HttpHeadersProgrammer } from "../../../programmers/http/HttpHeadersProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createHeaders",
      write: HttpHeadersProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/CreateHttpHeadersTransformer.go
package http

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type createHttpHeadersTransformerNamespace struct{}

var CreateHttpHeadersTransformer = createHttpHeadersTransformerNamespace{}

func (createHttpHeadersTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "http.createHeaders",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativehttpprogrammers.HttpHeadersProgrammer.Write(x)
    },
  })
}
